plugins/@grouparoo/closeio/src/lib/export-contacts/exportRecord.ts

```diff
@@ -0,0 +1,116 @@
+import { Errors, ExportRecordPluginMethod } from "@grouparoo/core";
+import { connect } from "../connect";
+import { CloseioCacheData } from "../common/destinationMappingOptions";
+import {
+  getExportRecordWithErrorHandling,
+  makePayload,
+} from "../common/exportRecord";
+
+export const exportContactRecord: ExportRecordPluginMethod = (arg) => {
+  return getExportRecordWithErrorHandling(handleContactChanges)(arg);
+};
+
+const handleContactChanges: ExportRecordPluginMethod = async ({
+  appId,
+  appOptions,
+  syncOperations,
+  export: {
+    newRecordProperties,
+    oldRecordProperties,
+    newGroups,
+    oldGroups,
+    toDelete,
+  },
+}) => {
+  if (Object.keys(newRecordProperties).length === 0) {
+    return { success: true };
+  }
+
+  if (!newRecordProperties["Name"]) {
```

Comment:
Is this always the primary key for a contact? I'd expect email to be a better primary key.

Reply:
Emails in contacts are of array type, it's not direct and searching with its value is misleading!
That's why I've used the name property.